### DISCUSSION TITLE
Improving the behavior and capabilities of tags

### DISCUSSION THREAD

**Discussion by David-van-der-Sluijs** [directus/directus#12590] [Feature Requests]
When you create keywords, they are listed according to the order in which they were entered and are deleted by a single click.

I find this behavior quite frustrating. For example, if I want to edit the tag at the very beginning I have to delete all tags including the tag I want to edit. Or if I want to change the order of the tags I also have to delete all tags.

Another problem is when I accidentally click on a tag it disappears immediately and I have to retype it.

So my suggestion would be that you can edit tags, reorder them & delete them less easily.

**Comment by benhaynes** [comment]
Love it! Some thoughts:

* Enable drag and drop of the tags to manually change the order
* Clicking a tag could overlay confirm/cancel icons
* Not sure about the "Edit". We could add an `edit` icon to the point above, but with the reordering feature added, it might be easier to delete, re-add, and reorder.

**Comment by David-van-der-Sluijs** [comment]
The edit feature will certainly come quite handy in my use case. I have tags which literally take up an entire row. Retyping the whole phrase just to edit a tiny detail is quite a chore.

I would imagine if I double clicked, the tag would turn into an input where I can make changes similar to Windows or macOS when you slowly double click a file/folder.

![2022-04-06 11 34 00](https://user-images.githubusercontent.com/26907265/161945140-5af29c89-d69e-45e7-95e4-eec68b98f9c3.gif)

This way it can be distinguished by the intention of only clicking once and then dragging to start rearranging the order.

To remove a tag a cross icon would appear next to the text once you hover over the tag. Clicking that icon a confirmation popup would appear asking if you'd really like to remove this tag. I also would add a checkbox to this dialog that would disable it, saying something in the likes of "Don't show this dialog again".

What do you think of this approach?

**Reply by benhaynes** [reply]
We'll have to be consistent with other interactions in the App, as we don't use double-click anywhere else. Since tags are a core component, this experience warrants a design sprint to see what will work best.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
